client/src/components/Pagination.tsx

```diff
@@ -0,0 +1,39 @@
+export type PaginationProps = {
+  itemsPerPage: number;
+  totalItems: number | undefined;
+  currentPage: number;
+  paginate: (number: number) => void;
```

Comment:
You should probably use a variable name that isn't a reserved word. Obviously, it works, but it can be confusing to the reader. I'd probably write it as `paginate: (page: number) => void;` Or should it be `paginate: (item: number) => void;`? As you can see, using just `number` is confusing, even without the fact that it's a reserved word.

Reply:
Ahh, that makes sense and would make the code a lot more clear.